### PR TITLE
#157845495 User should be able to add resource to a collection

### PR DESCRIPTION
**Pull request by mikey2020**
#### What does this PR do?
It adds a feature that allows users to add a resource to a collection

#### Description of completed tasks
- Add a handler for adding a resource to a collection
- Add endpoint `api/v1/collection/add/collection-id`
- Add tests for this feature

#### How should this be manually tested?
By hitting this endpoint `api/v1/collection/add/collection-id` with a POST request and a valid resource ID in the request body.

#### What are the relevant pivotal tracker stories?
https://www.pivotaltracker.com/story/show/157845495

### CODE REVIEW

File: handler/collection.go [hunk 1]
```diff
@@ -121,5 +122,57 @@ func (h *Handler) UpdateCollectionEndPoint(w http.ResponseWriter, r *http.Reques
...
+	if err != nil || payload.ResourceId == 0 {
+		utils.RespondWithError(w, http.StatusBadRequest,
+			"A valid ResourceId is required",
+		)
+		return
+	}
+
+	collectionId, _ := strconv.ParseInt(mux.Vars(r)["collectionId"], 10, 64)
+	query := `INSERT INTO resource_collections(resource_id, collection_id) VALUES ($1, $2)`
+	query = strings.TrimSuffix(query, ",")
```
Comment (tomipaul) [inline handler/collection.go]: @mikey2020 why are you trimming suffix here? I don't see a trailing comma.

File: utils_test.go [hunk 2]
```diff
@@ -36,12 +36,10 @@ var dummyData = map[string]interface{}{
...
 	},
 	"collection1": map[string]interface{}{
-		"name":   "first collection",
-		"userId": 1,
+		"name": "first collection",
 	},
 	"collection2": map[string]interface{}{
-		"name":   "second collection",
-		"userId": 1,
+		"name": "second collection",
```
Comment (tomipaul) [inline utils_test.go:42]: @mikey2020 In the dummy, there are four test collections. It's not paramount but I think we can do a bit of clean-up there 'cause we clearly don't need them all

**Comment by tomipaul** [comment]
@mikey2020 please address the concerns I raised so I can merge

File: collection_test.go [hunk 3]
```diff
@@ -264,5 +268,99 @@ func TestUpdateCollection(t *testing.T) {
...
+	t.Run("cannot be added because it is invalid", func(t *testing.T) {
+		Request(testServer.URL, t).
+			Post(collectionURI).
+			Set("authorization", userToken).
+			Send(invalidPayload).
+			Expect(400).
+			Expect("Content-Type", "application/json").
+			Expect(`{"error": "A valid ResourceId is required"}`).
+			End()
+	})
```
Comment (tomipaul) [inline collection_test.go:379]: @mikey2020 can you add a test case for a non-existing resource?

**Comment by mikey2020** [comment]
@tomipaul all changes have been made, please review

**Pull request merged**